tests/TestHelpers/HttpRequestHelper.php

```diff
@@ -0,0 +1,622 @@
+<?php declare(strict_types=1);
+/**
+ * ownCloud
+ *
+ * @author Artur Neumann <artur@jankaritech.com>
+ * @copyright Copyright (c) 2017 Artur Neumann artur@jankaritech.com
+ *
+ * This code is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License,
+ * as published by the Free Software Foundation;
+ * either version 3 of the License, or any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>
+ *
+ */
+
+namespace TestHelpers;
+
+use Exception;
+use GuzzleHttp\Client;
+use GuzzleHttp\Cookie\CookieJar;
+use GuzzleHttp\Exception\GuzzleException;
+use GuzzleHttp\Exception\RequestException;
+use GuzzleHttp\Psr7\Request;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\StreamInterface;
+use SimpleXMLElement;
+use Sabre\Xml\LibXMLException;
+use Sabre\Xml\Reader;
+use GuzzleHttp\Pool;
+
+/**
+ * Helper for HTTP requests
+ */
+class HttpRequestHelper {
+	public const HTTP_TOO_EARLY = 425;
+
+	/**
+	 * @var string
+	 */
+	private static $oCSelectorCookie = null;
+
+	/**
+	 * @return string
+	 */
+	public static function getOCSelectorCookie(): string {
+		return self::$oCSelectorCookie;
+	}
+
+	/**
+	 * @param string $oCSelectorCookie	"owncloud-selector=oc10;path=/;"
+	 *
+	 * @return void
+	 */
+	public static function setOCSelectorCookie(string $oCSelectorCookie): void {
+		self::$oCSelectorCookie = $oCSelectorCookie;
+	}
+
+	/**
+	 * Some systems-under-test do async post-processing of operations like upload,
+	 * move etc. If a client does a request on the resource before the post-processing
+	 * is finished, then the server should return HTTP_TOO_EARLY "425". Clients are
+	 * expected to retry the request "some time later" (tm).
+	 *
+	 * On such systems, when HTTP_TOO_EARLY status is received, the test code will
+	 * retry the request at 1-second intervals until either some other HTTP status
+	 * is received or the retry-limit is reached.
+	 *
+	 * @return int
+	 */
+	public static function numRetriesOnHttpTooEarly():int {
+		if (OcisHelper::isTestingOnOcisOrReva()) {
```

Comment:
Do we need such type of checks `isTestingOnOcisOrReva`? I think these tests now run only for ocis, I might be wrong.

Reply:
For this PR I'm just doing the copying things and true these code may not be necessary for tests anymore. But such filtration can be done in separate PR?

Reply:
Yes, we want to just "copy everything" for this PR (as much as possible) so that, in future, we can more easily track what happened. Functions like this can be refactored straight after this PR is merged. This function looks like it can always return `10`.